site/src/pages/TemplateSettingsPage/TemplateVariablesPage/TemplateVariablesPage.tsx

```diff
@@ -6,40 +6,103 @@ import {
 } from "api/typesGenerated";
 import { displaySuccess } from "components/GlobalSnackbar/utils";
 import { useOrganizationId } from "hooks/useOrganizationId";
-import { FC } from "react";
+import { useEffect, type FC, useState } from "react";
 import { Helmet } from "react-helmet-async";
 import { useNavigate, useParams } from "react-router-dom";
-import { templateVariablesMachine } from "xServices/template/templateVariablesXService";
-import { pageTitle } from "../../../utils/page";
+import { pageTitle } from "utils/page";
 import { useTemplateSettings } from "../TemplateSettingsLayout";
 import { TemplateVariablesPageView } from "./TemplateVariablesPageView";
+import {
+  createTemplateVersion,
+  templateVersion,
+  templateVersionVariables,
+  updateActiveTemplateVersion,
+} from "api/queries/templates";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { ErrorAlert } from "components/Alert/ErrorAlert";
+import { Loader } from "components/Loader/Loader";
 
 export const TemplateVariablesPage: FC = () => {
   const { template: templateName } = useParams() as {
     organization: string;
     template: string;
   };
-  const organizationId = useOrganizationId();
+  const orgId = useOrganizationId();
   const { template } = useTemplateSettings();
   const navigate = useNavigate();
-  const [state, send] = useMachine(templateVariablesMachine, {
-    context: {
-      organizationId,
-      template,
-    },
-    actions: {
-      onUpdateTemplate: () => {
-        displaySuccess("Template updated successfully");
+  const queryClient = useQueryClient();
+  const versionId = template.active_version_id;
+
+  const [newVersionId, setNewVersionId] = useState<string | null>(null);
+
+  const {
+    data: version,
+    error: versionError,
+    isLoading: isVersionLoading,
+  } = useQuery({ ...templateVersion(versionId), keepPreviousData: true });
+  const {
+    data: variables,
+    error: variablesError,
+    isLoading: isVariablesLoading,
+  } = useQuery({
+    ...templateVersionVariables(versionId),
+    keepPreviousData: true,
+  });
+
+  const createVersionMutation = createTemplateVersion(orgId, queryClient);
+  const { mutate: sendCreateTemplateVersion, error: createVersionError } =
+    useMutation({
+      ...createVersionMutation,
+      onSuccess: (data) => {
+        setNewVersionId(data.id);
       },
+    });
+  const publishMutation = updateActiveTemplateVersion(template, queryClient);
+  const { mutate: publishVersion, error: publishError } = useMutation({
+    ...publishMutation,
+    onSuccess: () => {
+      publishMutation.onSuccess();
+      displaySuccess("Template updated successfully");
```

Comment:
I think would be best to handle this in the handler and use `mutateAsync`. I think we had a conversation to define this on Slack.

https://codercom.slack.com/archives/CJURPL8DN/p1694547382069569?thread_ts=1694547252.468949&cid=CJURPL8DN

Reply:
I guess I didn't entirely internalize that conversation :p